Fix `RolloutAnyscaleService` after Anyscale SDK breaking change

tatiana opened the pull request:
The Anyscale Python SDK introduced a breaking change by removing support for the `Service` `config` argument and introducing the `configs` argument. I suspect this happened somewhere between versions 0.24.54 (the provider is pinned to this version or higher) and 0.26.75 (the last released version). However, since the tests were not running as expected, and there is no changelog for the Anyscale Python SDK, it is hard to pinpoint when the change happened.

The new interface is described in the official documentation: https://docs.anyscale.com/reference/service-api\#service-sdk
<img width="902" height="605" alt="Screenshot 2025-11-20 at 11 29 04" src="https://github.com/user-attachments/assets/e8033a02-223b-4884-acf8-50d5c56b29e4" />

This breaking change led to the `RolloutAnyscaleService` operator stopping working.

This issue was identified while working on https://github.com/astronomer/astro-provider-anyscale/pull/92, when I improved how the integration tests work, and I noticed that the example DAG `sample_anyscale_service_workflow` has been failing with the following error message:
```
  File "/Users/tatiana.alchueyr/Code/astro-provider-anyscale/anyscale_provider/operators/anyscale.py", line 385, in execute
    self.service_id = self.hook.deploy_service(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tatiana.alchueyr/Code/astro-provider-anyscale/anyscale_provider/hooks/anyscale.py", line 99, in deploy_service
    service_id: str = self.client.service.deploy(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: ServiceSDK.deploy() got an unexpected keyword argument 'config'
```

This PR solves the issue by:
- deprecating the argument `config` in the `AnyscaleHook.deploy_service` method
- introducing a new argument `configs`  in the `AnyscaleHook.deploy_service` method
- handling errors generated in earlier versions of the SDK

I observed that after this fix, there is a different issue in the CI, which will be addressed in a separate PR:
```
[2025-11-20 16:34:21,609] {anyscale.py:420} INFO - Execution completed for service AstroService-CICD-test-94-3-11-2-9 with state: SYSTEM_FAILURE
[2025-11-20T16:34:21.609+0000] {anyscale.py:420} INFO - Execution completed for service AstroService-CICD-test-94-3-11-2-9 with state: SYSTEM_FAILURE
INFO     airflow.task.operators.anyscale_provider.operators.anyscale.RolloutAnyscaleService:anyscale.py:420 Execution completed for service AstroService-CICD-test-94-3-11-2-9 with state: SYSTEM_FAILURE
[2025-11-20 16:34:21,610] {anyscale.py:425} ERROR - Anyscale service deployment AstroService-CICD-test-94-3-11-2-9 failed with error: asyncio.run() cannot be called from a running event loop
[2025-11-20T16:34:21.610+0000] {anyscale.py:425} ERROR - Anyscale service deployment AstroService-CICD-test-94-3-11-2-9 failed with error: asyncio.run() cannot be called from a running event loop
ERROR    airflow.task.operators.anyscale_provider.operators.anyscale.RolloutAnyscaleService:anyscale.py:425 Anyscale service deployment AstroService-CICD-test-94-3-11-2-9 failed with error: asyncio.run() cannot be called from a running event loop
[2025-11-20T16:34:21.610+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
INFO     airflow.models.taskinstance:taskinstance.py:441 ::group::Post task execution logs
[2025-11-20 16:34:21,614] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/runner/.local/share/hatch/env/virtual/astro-provider-anyscale/UDnurkOL/tests.py3.11-2.9/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.local/share/hatch/env/virtual/astro-provider-anyscale/UDnurkOL/tests.py3.11-2.9/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.local/share/hatch/env/virtual/astro-provider-anyscale/UDnurkOL/tests.py3.11-2.9/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 1700, in resume_execution
    return execute_callable(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/astro-provider-anyscale/astro-provider-anyscale/anyscale_provider/operators/anyscale.py", line 426, in execute_complete
    raise AirflowException(error_msg)
airflow.exceptions.AirflowException: Anyscale service deployment AstroService-CICD-test-94-3-11-2-9 failed with error: asyncio.run() cannot be called from a running event loop
```

tatiana committed:
Fix RolloutAnyscaleService

The Anyscale SDK introduced a breaking change, by renaming in the Anyscale SDK the Service argument config to the plural versionc onfigs.

Latest version documented in: https://docs.anyscale.com/reference/service-api\#service-sdk

This lead to the RolloutAnyscaleService operator to stop working

tatiana committed:
Revert change to integration_test.sh

tatiana committed:
Fix unit tests

pankajkoti reviewed (approved):
Looks good, but have a suggestion if you'd like to consider. No objection if you'd like to go ahead as is.

tatiana committed:
Support latest version of the Anyscale SDK while not introducing a breaking change

tatiana committed:
Add Cursor generated tests to cover areas of the hook that were introduced and were lacking tests

tatiana committed:
Merge branch 'main' into fix-RolloutAnyscaleService-operator

tatiana committed:
Try to get the AnyscaleHook to be able to deploy a service again.

Trying to tackle the failure:
```
  File "/home/runner/work/astro-provider-anyscale/astro-provider-anyscale/anyscale_provider/operators/anyscale.py", line 380, in execute
    self.service_id = self.hook.deploy_service(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/astro-provider-anyscale/astro-provider-anyscale/anyscale_provider/hooks/anyscale.py", line 112, in deploy_service
    service_id: str = self.client.service.deploy(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.local/share/hatch/env/virtual/astro-provider-anyscale/UDnurkOL/tests.py3.11-2.9/lib/python3.11/site-packages/anyscale/service/__init__.py", line 87, in deploy
    return self._private_sdk.deploy(
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.local/share/hatch/env/virtual/astro-provider-anyscale/UDnurkOL/tests.py3.11-2.9/lib/python3.11/site-packages/anyscale/service/_private/service_sdk.py", line 736, in deploy
    raise ValueError(
ValueError: --versions must be provided when deploying with multiple config files.
```

Observed in the CI:
https://github.com/astronomer/astro-provider-anyscale/actions/runs/19542809657/job/55953570162?pr=94

tatiana committed:
Fix error related to --versions
Example: https://github.com/astronomer/astro-provider-anyscale/actions/runs/19543062951/job/55954508112"

```
  File "/home/runner/work/astro-provider-anyscale/astro-provider-anyscale/anyscale_provider/operators/anyscale.py", line 380, in execute
    self.service_id = self.hook.deploy_service(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/astro-provider-anyscale/astro-provider-anyscale/anyscale_provider/hooks/anyscale.py", line 112, in deploy_service
    service_id: str = self.client.service.deploy(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.local/share/hatch/env/virtual/astro-provider-anyscale/UDnurkOL/tests.py3.11-2.9/lib/python3.11/site-packages/anyscale/service/__init__.py", line 87, in deploy
    return self._private_sdk.deploy(
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.local/share/hatch/env/virtual/astro-provider-anyscale/UDnurkOL/tests.py3.11-2.9/lib/python3.11/site-packages/anyscale/service/_private/service_sdk.py", line 736, in deploy
    raise ValueError(
ValueError: --versions must be provided when deploying with multiple config files.
```

tatiana commented:
@pankajastro @pankajkoti thanks for your feedback! I ended up making the change backwards-compatible..!

tatiana commented on anyscale_provider/hooks/anyscale.py:
```diff
@@ -77,7 +77,7 @@ def submit_job(self, config: JobConfig) -> str:
     def deploy_service(
         self,
-        config: ServiceConfig,
+        configs: list[ServiceConfig],
```
I ended up implementing the change so it is backwards compatible, so no need to add notes to the changelog
